app/scenes/channel_info/channel_info.js

```diff
@@ -71,14 +73,23 @@ class ChannelInfo extends PureComponent {
         if (isFavorite !== this.state.isFavorite) {
             this.setState({isFavorite});
         }
-        this.navigateAfterLeave(nextProps.leaveChannelRequest);
+        this.navigateAfterLeave(nextProps.leaveChannelRequest, nextProps.deleteChannelRequest);
     }
 
-    navigateAfterLeave(leaveChannelRequest) {
-        if (
+    deleteOrLeaveRequestSuccessful(leaveChannelRequest, deleteChannelRequest) {
```

Comment:
Could this also be done in the action/view where deleteChannel and leaveChannel is fired? Looks like the deleteChannel and leaveChannel in the service/actions dispatch on failed so could you watch the request status there instead of in the view?